src/builtin-templates/get-markup.js

```diff
@@ -0,0 +1,39 @@
+import { html } from '/web_modules/lit-html/lit-html.js';
+import { unsafeHTML } from '/web_modules/lit-html/directives/unsafe-html.js';
+
+import priceDisplay from './price-display.js';
+import file from './file-download.js';
+
+function markup(item) {
+    if (item.type === 'StructuredText') {
+        return html`
+            <article>
+                <header>
+                    <p><b>${item.name}</b></p>
+                </header>
+                <div class="dim">${item.contents.map(markup)}</div>
+            </article>`;
+    }
+
+    if (item.type === 'StructuredPrice') {
+        return html`
+            <article>
+                <header>
+                    <p><b>${item.name}</b><br />${priceDisplay(item.price)}</p>
+                </header>
+                <div class="dim">${item.contents.map(markup)}</div>
+            </article>`;
+    }
+
+    switch (item.type) {
```

Comment:
I'm not sure about having ifs and a switch. I guess you've reserved the switch here for one-liners.

Reply:
yup, exactly. Is it weird? I'll rewrite with ifs only if you suggest so!

Reply:
No need, just checking my assumptions.